pkg/sql/sqlbase/table.go

```diff
@@ -196,13 +196,6 @@ func MakeColumnDefDescs(
 		if typedExpr, err = t.NormalizeExpr(evalCtx, typedExpr); err != nil {
 			return nil, nil, err
 		}
-		// Try to evaluate once. If it is aimed to succeed during a
-		// backfill, it must succeed here too. This tries to ensure that
-		// we don't end up failing the evaluation during the schema change
-		// proper.
-		if _, err := typedExpr.Eval(evalCtx); err != nil {
-			return nil, nil, err
-		}
```

Comment:
this was introduced in https://github.com/cockroachdb/cockroach/pull/15345 . Do you think you can disable it for some types of expressions?

Reply:
I think that "check by evaluation" is conceptually flawed. 
1) there are expressions that just cannot be checked this way -- they will only cause an error in some evaluations, so even if we have this check it doesn't guarantee that the backfill will succeed
2) as Pete says it prevents us from having expressions with side effects. `nextval` is our first instance of this, but if/when we support user-defined functions/procedures with side effects, "evaluation for checking" will be simply unacceptable.

I think we should simply accept the expression as-is if it passes type checking, and instead be robust with erroring out on backfills and allowing the user to replace it with ALTER.